Sources/Brave/WebFilters/Debug/TestAdblockEnginesView.swift

```diff
@@ -0,0 +1,255 @@
+                filterListInfo: info.value.filterListInfo, isAlwaysAggressive: info.value.isAlwaysAggressive
+              )
+              numberOfSharedEngines = await AdBlockStats.shared.numberOfEngines
+              self.compiledSharedEngineCount = (run * infos.count) + (index + 1)
+              try await Task.sleep(for: .milliseconds(delay))
+            }
+          }
+        }
+          
+        compileSharedEnginesResult = .success(result)
+      } catch {
+        compileSharedEnginesResult = .failure(error)
+      }
+      
+      compileSharedEnginesRunning = false
+    }
+  }
+  
+  private func startTest() {
+    adblockStats = AdBlockStats()
+    testStarted = true
+    compiledCount = 0
+    result = nil
+    testRunning = true
+    
+    Task {
+      do {
+        let clock = ContinuousClock()
+        let result = try await clock.measure {
+          switch compileType {
+          case .series:
+            try await (0..<numberOfEngines).asyncForEach { index in
+              try await compileEngine(at: index, compileType: compileType)
+              self.compiledCount = index + 1
+            }
+          case .parallel:
+            try await (0..<numberOfEngines).asyncConcurrentForEach { index in
+              try await compileEngine(at: index, compileType: compileType)
+              self.compiledCount += 1
+            }
+          }
+          
+        }
+        self.result = .success(result)
+      } catch {
+        self.result = .failure(error)
+      }
+      
+      testRunning = false
+    }
+  }
+  
+  private func compileEngine(at index: Int, compileType: CompileType) async throws {
+    let uuid = "test-engine-\(index)"
+    let sampleFilterListURL = Bundle.module.url(forResource: "list", withExtension: "txt")!
+    let resourcesURL = Bundle.module.url(forResource: "resources", withExtension: "json")!
+    
+    let resourcesInfo = CachedAdBlockEngine.ResourcesInfo(
+      localFileURL: resourcesURL, version: "1.0"
+    )
+    
+    let filterListInfo = CachedAdBlockEngine.FilterListInfo(
+      source: .filterList(componentId: uuid),
+      localFileURL: sampleFilterListURL,
+      version: "bundled", fileType: .text
+    )
+    
+    switch compileType {
+    case .parallel:
+      try await adblockStats.compileAsync(
+        filterListInfo: filterListInfo, resourcesInfo: resourcesInfo, isAlwaysAggressive: false
+      )
+      
+      if removeEngineAfterCompilation {
+        Task.detached {
+          await adblockStats.removeEngine(for: .filterList(componentId: uuid))
+        }
+      }
+    case .series:
+      try await adblockStats.compile(
+        filterListInfo: filterListInfo, resourcesInfo: resourcesInfo, isAlwaysAggressive: false
+      )
+      
+      if removeEngineAfterCompilation {
+        await adblockStats.removeEngine(for: .filterList(componentId: uuid))
+      }
+    }
+  }
+}
+
+#if swift(>=5.9)
+@available(iOS 17.0, *)
+#Preview {
+  TestAdblockEnginesView()
+}
+#endif
```

Comment:
Debugging tool to brute force large engine compilations under different conditions